samza-core/src/main/java/org/apache/samza/storage/NonTransactionalStateTaskRestoreManager.java

```diff
@@ -110,20 +110,21 @@
    */
   @Override
   public void init(Map<SystemStreamPartition, String> checkpointedChangelogSSPOffsets) {
-    cleanBaseDirsAndReadOffsetFiles();
+    cleanBaseDirsAndReadOffsetFiles(new StorageConfig(config).getCleanLoggedStoreDirsOnInit());
```

Comment:
What about the other restore manager?

Extract variables.

Add tests.

Reply:
The TransactionalStateTaskRestoreManager already seems to delete storeDirs on start after reading?

Reply:
It always deletes the `current` directory, but may restore from a store snapshot/checkpoint. So you'll either need to disable restore completely, or clear the restored store at the end.

Reply:
done